src/python/pants/backend/go/util_rules/external_module.py

```diff
@@ -178,80 +174,54 @@ async def resolve_external_go_package(
 
 
 @dataclass(frozen=True)
-class PackagesFromExternalModuleRequest:
+class ExternalModulePkgImportPathsRequest:
+    """Request the import paths for all packages belonging to an external Go module.
+
+    The `go_sum_digest` must have a `go.sum` file that includes the module and its dependencies,
+    else the user will get a `missing go.sum entry` error.
+    """
+
     module_path: str
     version: str
     go_sum_digest: Digest
 
 
-class PackagesFromExternalModule(DeduplicatedCollection[ResolvedGoPackage]):
-    pass
+class ExternalModulePkgImportPaths(DeduplicatedCollection[str]):
+    """The import paths for all packages belonging to an external Go module."""
+
+    sort_input = True
 
 
 @rule
-async def compute_packages_from_external_module(
-    request: PackagesFromExternalModuleRequest,
-) -> PackagesFromExternalModule:
+async def compute_package_import_paths_from_external_module(
+    request: ExternalModulePkgImportPathsRequest,
+) -> ExternalModulePkgImportPaths:
     module_path = request.module_path
     module_version = request.version
 
     downloaded_module = await Get(
-        DownloadedExternalModule,
-        DownloadExternalModuleRequest(module_path, module_version),
+        DownloadedExternalModule, DownloadExternalModuleRequest(module_path, module_version)
     )
-    sources_digest = await Get(Digest, AddPrefix(downloaded_module.digest, "__sources__"))
-
-    # TODO: Super hacky merge of go.sum from both digests. We should really just pass in the fully-resolved
-    # go.sum and use that, but this allows the go.sum from the downloaded module to have some effect. Not sure
-    # if that is right call, but hackity hack!
-    left_digest_contents = await Get(DigestContents, Digest, sources_digest)
-    left_go_sum_contents = b""
-    for fc in left_digest_contents:
-        if fc.path == "__sources__/go.sum":
-            left_go_sum_contents = fc.content
-            break
-
-    go_sum_prefixed_digest = await Get(Digest, AddPrefix(request.go_sum_digest, "__sources__"))
-    right_digest_contents = await Get(DigestContents, Digest, go_sum_prefixed_digest)
-    right_go_sum_contents = b""
-    for fc in right_digest_contents:
-        if fc.path == "__sources__/go.sum":
-            right_go_sum_contents = fc.content
-            break
-    go_sum_contents = left_go_sum_contents + b"\n" + right_go_sum_contents
-    go_sum_digest = await Get(
-        Digest, CreateDigest([FileContent("__sources__/go.sum", go_sum_contents)])
+    downloaded_digest_without_go_sum = await Get(
+        Digest, DigestSubset(downloaded_module.digest, PathGlobs(["**", "!go.sum"]))
     )
```

Comment:
This is still weird, I think probably `DownloadedExternalModule.digest` should proactively exclude `go.sum`. But I tried to keep this diff small because I don't yet understand how `go.sum` interacts with `build_go_pkg.py`.